src/main/java/org/kuali/kfs/module/ar/document/web/struts/PaymentApplicationAdjustmentAction.java

```diff
@@ -0,0 +1,1291 @@
+import org.kuali.kfs.krad.util.GlobalVariables;
+import org.kuali.kfs.krad.util.KRADConstants;
+import org.kuali.kfs.krad.util.KRADPropertyConstants;
+import org.kuali.kfs.krad.util.MessageMap;
+import org.kuali.kfs.krad.util.ObjectUtils;
+import org.kuali.kfs.krad.util.UrlFactory;
+import org.kuali.kfs.module.ar.ArKeyConstants.PaymentApplicationAdjustmentDocumentErrors;
+import org.kuali.kfs.module.ar.businessobject.AccountsReceivableDocumentHeader;
+import org.kuali.kfs.module.ar.businessobject.Customer;
+import org.kuali.kfs.module.ar.businessobject.InvoicePaidApplied;
+import org.kuali.kfs.module.ar.document.CustomerInvoiceDocument;
+import org.kuali.kfs.module.ar.document.PaymentApplicationAdjustmentDocument;
+import org.kuali.kfs.module.ar.document.PaymentApplicationDocument;
+import org.kuali.kfs.module.ar.document.service.AccountsReceivableDocumentHeaderService;
+import org.kuali.kfs.module.ar.document.service.impl.PaymentApplicationAdjustmentDocumentService;
+import org.kuali.kfs.module.ar.rest.resource.requests.AdHocRoutingRequest;
+import org.kuali.kfs.module.ar.rest.resource.requests.PaymentApplicationAdjustmentRequest;
+import org.kuali.kfs.module.ar.rest.resource.requests.PaymentApplicationAdjustmentRequest.AccountingLine;
+import org.kuali.kfs.module.ar.rest.resource.responses.PaymentApplicationAdjustmentInvoiceResponse;
+import org.kuali.kfs.sys.KFSConstants;
+import org.kuali.kfs.sys.Message;
+import org.kuali.kfs.sys.MessageBuilder;
+import org.kuali.kfs.sys.businessobject.SourceAccountingLine;
+import org.kuali.kfs.sys.context.SpringContext;
+import org.kuali.kfs.sys.document.service.AccountingLineRuleHelperService;
+import org.kuali.kfs.sys.document.service.impl.AccountingLineValidationError;
+import org.kuali.kfs.sys.document.web.struts.FinancialSystemTransactionalDocumentActionBase;
+import org.kuali.kfs.sys.rest.resource.responses.ErrorResponse;
+import org.kuali.rice.core.api.exception.RiceIllegalArgumentException;
+import org.kuali.rice.core.api.util.type.KualiDecimal;
+import org.kuali.rice.kew.api.action.ActionRequest;
+import org.kuali.rice.kew.api.action.InvalidActionTakenException;
+import org.kuali.rice.kew.api.document.InvalidDocumentContentException;
+import org.kuali.rice.kew.api.exception.WorkflowException;
+import org.kuali.rice.kim.api.identity.Person;
+import org.springframework.http.HttpHeaders;
+import org.springframework.util.AutoPopulatingList;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Map;
+import java.util.List;
+import java.util.Objects;
+import java.util.HashMap;
+import java.util.Set;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.core.Response;
+
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static java.util.Map.entry;
+
+/**
+ * CU Customization KFSPTS-21426 - FINP 7302 - APPA GLPEs are not created correctly
```

Comment:
I think this is a backport rather than a customization?

Reply:
how about KFSPTS-21426 - backport FINP 7302

Reply:
sounds good